src/subprocess-posix.cc

```diff
@@ -330,9 +351,81 @@ void SubprocessSet::Clear() {
     // Since the foreground process is in our process group, it will receive
     // the interruption signal (i.e. SIGINT or SIGTERM) at the same time as us.
     if (!(*i)->use_console_)
-      kill(-(*i)->pid_, interrupted_);
+      Xkill(-(*i)->pid_, interrupted_);
   for (vector<Subprocess*>::iterator i = running_.begin();
        i != running_.end(); ++i)
     delete *i;
   running_.clear();
 }
+
+void SubprocessSet::Suspend() {
+  assert(IsSuspended());
+
+  // errno may change during the course of this function.
+  int saved_errno = errno;
+
+  // Suspend children.
+  for (vector<Subprocess*>::iterator i = running_.begin();
+       i != running_.end(); ++i)
+    if (!(*i)->use_console_)
+      Xkill(-(*i)->pid_, SIGTSTP);
+
+  // Tell users what we have done.
+  {
+    // We use write to bypass the buffer that may not be flush before
+    // we re-raise the SIGTSTP for us.
```

Comment:
I have tested on OSX Sierra 10.12.6 and Ubuntu 16.04.3 and the message appears after suspending and before resuming as expected. 

The buffer I am speaking about is the internal buffer that is used by `printf` or `fprintf`. That's why I use `write(1)` directly. I can update the comment if it is not clear.

Here what I have when testing:
```
$ ./ninja -f misc/long-slow-build.ninja
[  0%] SLEEP 5^Z
ninja: entire build suspended.

zsh: suspended  ./ninja -f misc/long-slow-build.ninja
$ fg
[1]  + continued  ./ninja -f misc/long-slow-build.ninja
[ 60%] SLEEP 23^Z
ninja: entire build suspended.

zsh: suspended  ./ninja -f misc/long-slow-build.ninja
$ fg
1]  + continued  ./ninja -f misc/long-slow-build.ninja
[100%] SLEEP 29
```